python/cudf/cudf/core/frame.py

```diff
@@ -2693,7 +2693,14 @@ def acos(self):
                     1.5707963267948966,  1.266103672779499],
                     dtype='float64')
         """
-        return self._unaryop("acos")
+        if self.iloc[0].dtype.kind in ("i", "u"):
```

Comment:
Indexing to a single element of a DataFrame / MultiIndex will possibly copy to a new Series, and indexing to a single element of a Series / Index currently copies the scalar to host (and synchronizes). Could we take a different approach here that doesn't require the indexing here?

Reply:
Makes sense, I was using indexing because I was getting an error when I tried to use dtype generally. The tests would throw an error for Dataframes and vice-versa if I tried to use dtypes. I think Prem's suggestion might help with this.

Reply:
oof, I just realised I can just use self._column, instead lol. Switching to that now XD